lib/paperclip/lazy_thumbnail.rb

```diff
@@ -1,24 +1,128 @@
 # frozen_string_literal: true
 
 module Paperclip
-  class LazyThumbnail < Paperclip::Thumbnail
+  class LazyThumbnail < Paperclip::Processor
+    GIF_FPS = 12
+
+    GIF_PALETTE_COLORS = 32
+
+    ALLOWED_FIELDS = %w(
+      icc-profile-data
+    ).freeze
+
+    class PixelGeometryParser
+      def self.parse(current_geometry, pixels)
+        width  = Math.sqrt(pixels * (current_geometry.width.to_f / current_geometry.height)).round.to_i
+        height = Math.sqrt(pixels * (current_geometry.height.to_f / current_geometry.width)).round.to_i
+
+        Paperclip::Geometry.new(width, height)
+      end
+    end
+
+    def initialize(file, options = {}, attachment = nil)
+      super
+
+      @crop = options[:geometry].to_s[-1, 1] == '#'
+      @current_geometry = options.fetch(:file_geometry_parser, Geometry).from_file(@file)
+      @target_geometry = options[:pixels] ? PixelGeometryParser.parse(@current_geometry, options[:pixels]) : options.fetch(:string_geometry_parser, Geometry).parse(options[:geometry].to_s)
+      @format = options[:format]
+      @current_format = File.extname(@file.path)
+      @basename = File.basename(@file.path, @current_format)
+
+      correct_current_format!
+    end
+
     def make
       return File.open(@file.path) unless needs_convert?
 
-      if options[:geometry]
-        min_side = [@current_geometry.width, @current_geometry.height].min.to_i
-        options[:geometry] = "#{min_side}x#{min_side}#" if @target_geometry.square? && min_side < @target_geometry.width
-      elsif options[:pixels]
-        width  = Math.sqrt(options[:pixels] * (@current_geometry.width.to_f / @current_geometry.height)).round.to_i
-        height = Math.sqrt(options[:pixels] * (@current_geometry.height.to_f / @current_geometry.width)).round.to_i
-        options[:geometry] = "#{width}x#{height}>"
+      dst = TempfileFactory.new.generate([@basename, @format ? ".#{@format}" : @current_format].join)
+
+      if preserve_animation?
+        if @target_geometry.nil? || (@current_geometry.width <= @target_geometry.width && @current_geometry.height <= @target_geometry.height)
+          target_width = 'iw'
+          target_height = 'ih'
+        else
+          scale = [@target_geometry.width.to_f / @current_geometry.width, @target_geometry.height.to_f / @target_geometry.height].min
```

Comment:
Typo? Shouldn't this be:
```suggestion
          scale = [@target_geometry.width.to_f / @current_geometry.width, @target_geometry.height.to_f / @current_geometry.height].min
```